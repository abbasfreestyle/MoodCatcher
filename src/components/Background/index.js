import React from 'react';
import PropTypes from 'prop-types';

import { StyleSheet } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import theme from 'theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

const Background = props => {
  const { children, ...rest } = props;
  return (
    <LinearGradient
      style={styles.container}
      colors={[theme.background, theme.surface]}
      {...rest}
    >
      {children}
    </LinearGradient>
  );
};

Background.propTypes = {
  children: PropTypes.node.isRequired
};

export default Background;
