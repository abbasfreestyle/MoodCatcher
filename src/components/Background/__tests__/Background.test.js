import React from 'react';
import { View, Text } from 'react-native';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Background from '..';

const requiredProps = {
  children: (
    <View>
      <Text>Test Content</Text>
    </View>
  )
};

const wrapper = shallow(<Background {...requiredProps} />);

describe('Background.', () => {
  describe('Snapshots.', () => {
    test('should match', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
