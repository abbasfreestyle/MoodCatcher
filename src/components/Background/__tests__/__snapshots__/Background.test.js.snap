// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background. Snapshots. should match 1`] = `
<LinearGradient
  colors={
    Array [
      "#FFFFFF",
      "#F2FCFB",
    ]
  }
  end={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  start={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text>
      Test Content
    </Text>
  </View>
</LinearGradient>
`;
